client/state/reader/conversations/reducer.js

```diff
@@ -0,0 +1,62 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_CONVERSATION_FOLLOW,
+	READER_CONVERSATION_MUTE,
+	READER_CONVERSATION_UPDATE_FOLLOW_STATUS,
+} from 'state/action-types';
+import {
+	CONVERSATION_FOLLOW_STATUS_FOLLOWING,
+	CONVERSATION_FOLLOW_STATUS_MUTING,
+} from './follow-status';
+import { combineReducers, createReducer, isValidStateWithSchema } from 'state/utils';
+import { itemsSchema } from './schema';
+import { key } from './utils';
+
+/**
+ * Tracks all known conversation following statuses.
+ */
+export const items = createReducer(
```

Comment:
i've been wondering what happens if we do something like:
```js
const items = keyedReducer( 'payload.siteId', 
  keyedReducer( payload.postId, createReducer( null, {
    [ READER_CONVERSATION_FOLLOW ]: () =>  CONVERSATION_FOLLOW_STATUS_FOLLOWING,
    [ READER_CONVERSATION_MUTE ]: () => CONVERSATION_FOLLOW_STATUS_MUTING,
    [ READER_CONVERSATION_UPDATE_STATUS ]: (state, action) => action.payload.followStatus 
} ) )

items.schema = itemsSchema;
```

this is such a common pattern would be nice if we had a smarter HoR that handled something like this. *cough* *cough* @jsnajdr

Reply:
Interesting! 

My first thought was to use `keyedReducer` with our compound key (`siteId-postId`) and the key has to exist in the payload already, so that wasn't possible.

Reply:
Your idea is definitely superior than needing to nest two keyedReducers. Would be nice if we could specify compounds keys like that